bot.go

```diff
@@ -269,7 +270,15 @@ func evalAndReply(session *discordgo.Session, info, channelID, input string,
 	replyID string, formatMode, compactMode, verbatimMode bool,
 ) string {
 	res := eval(input, formatMode, compactMode, verbatimMode)
-	log.S(log.Info, info, log.String("response", res))
+	level := log.Info
+	msg := "response"
+	if len(res) > 1900 {
+		// hope we didn't garbage some unicode char in the middle of a utf8 char...
```

Comment:
You can use runes for that.

https://github.com/cdfmlr/ellipsis/blob/main/ellipsis.go#L51-L61

Another solution is to look for the last space before the 1900th character, then cut and add the ellipsis.

This doesn't work for continuous text like CJK

A common way is too look for the last space character between the 1800th and 1900th characters, if none is found you can apply your strategy. But using runes should be prefered.

Reply:
Another solution

https://pkg.go.dev/tailscale.com/util/truncate#String

Reply:
I know I can use runes :) just that I didn't feel like paying the overhead

this being said allegedly the limit is in characters/runes... so I will try and confirm if that's true

and I did test it and discord handles bad utf8 fine (they kinda have to)

<img width="329" alt="Screenshot 2024-08-16 at 4 16 25 PM" src="https://github.com/user-attachments/assets/f6dc2375-1291-458a-a2e1-934a1448ac4a">

Reply:
fixed in #79